cfme/scripting/requirement.py

```diff
@@ -0,0 +1,336 @@
+import re
+import subprocess
+import sys
+from pprint import PrettyPrinter
+
+import click
+import yaml
+
+from cfme.utils.path import get_rel_path
+from cfme.utils.path import project_path
+from cfme.utils.path import requirements_path
+from requirements import RepoImportScanner
+
+
+# Check for default template exists, its passed to freeze if so
+# if user passes bad path for --extra-template, freeze will hit RuntimeError
+DEFAULT_EXTRA_TEMPLATE = requirements_path.join("template_non_imported.txt")
+
+# These files is written out to, don't have to assert it exists yet
+DEFAULT_SCAN_TEMPLATE = requirements_path.join("template_scanned_imports.txt")
+DEFAULT_FROZEN_OUTPUT = requirements_path.join("frozen.txt")
+DEFAULT_CONSTRAINT_FILE = requirements_path.join("constraints.txt")
+
+# Package map
+DEFAULT_PACKAGE_MAP_FILE = requirements_path.join("package_map.yaml")
+
+
+# ---- utility functions --------------
+def get_package_map(package_map_file):
+    package_map = None
+    try:
+        with open(package_map_file, 'r') as f:
+            package_map = yaml.safe_load(f).get("packages")
+    except FileNotFoundError:
+        pass
+    return package_map
+
+
+def remove_dup_packages(package_list, package):
+    # get name from package
+    name = re.split("[=><]", package)[0]
+    # turn package_list into list of package_names
+    package_names = [
+        re.split("[=><]", package_name)[0] for package_name in package_list
+    ]
+    # get rid of duplicate names
+    duplicate_indices = [
+        i for i, package_name in enumerate(package_names) if package_name == name
+    ]
+    # get rid of the first occurrence since the second will be the recently appended item
+    if len(duplicate_indices) > 1:
+        del package_list[duplicate_indices[0]]
```

Comment:
**Question** what happens if there are more than 2 duplicates in the `package_list` ?

Reply:
Then it will remove the first occurrence of the package name. However, there should never be more than 2 duplicates in `package_list` (unless someone has added a package manually to a file). This is a helper function for `add_package_to_file` so that a package does not appear twice in the package list file. I can make it so that it removes all but the last occurrence of the package name in `package_list`, that may be safer. What do you think?

Reply:
I think that might help. Right now what you have feels like an assumption, but if we check and make sure of it then its no more a guess.